design/src/rfcs/rfc0021_dependency_versions.md

```diff
@@ -0,0 +1,71 @@
+RFC: Dependency Versions
+========================
+
+> Status: RFC
+>
+> Applies to: Client and Server
+
+This RFC outlines how Rust dependency versions are selected for the smithy-rs project, and
+strives to meet the following semi-conflicting goals:
+
+- Dependencies are secure
+- Vended libraries have dependency ranges that overlap other Rust libraries as much as possible
+
+When in conflict, the security goal takes priority over the compatibility goal.
+
+Categorization of Crates
+------------------------
+
+The Rust crates within smithy-rs can be divided up into two categories:
+
+1. **Library Crates:** Crates that are published to crates.io with the intention that other projects
+   will depend on them via their `Cargo.toml` files. This category does NOT include binaries that
+   are published to crates.io with the intention of being installed with `cargo install`.
+2. **Application Crates:** All examples, binaries, tools, standalone tests, or other crates that are
+   not published to crates.io with the intent of being depended on by other projects.
+
+### Code generated crates
+
+Generated clients must be considered library crates even if they're not published since they are intended
+to be pulled into other Rust projects with other dependencies.
+
+Generated servers must be considered application crates.
```

Comment:
Why? I don't understand the distinction. The sentence above for generated clients applies for generated servers too.

Reply:
Yeah, actually, good point. The server runtime and code generated crates need to be broadly compatible with the libraries used by the server implementers consuming them. I'll fix this.